frontend/src/pages/details-page.tsx

```diff
@@ -12,6 +17,8 @@ export default function DetailsPage() {
   const [loading, setIsLoading] = useState(initialVal);
   const { postId } = useParams();
   const navigate = useNavigate();
+  const [categoryPosts, setcategoryPosts] = useState<Post[]>([]);
+  const [postLoading, setpostLoading] = useState(false);
 
```

Comment:
I see this is declared, but never used anywhere, ideally this should be used and the naming should be relatedPostsLoading and shimmer view needs to be provided when it is loading